subiquity/server/controllers/filesystem.py

```diff
@@ -377,11 +380,59 @@ async def guided_GET(self, wait: bool = False) -> GuidedStorageResponse:
             disks=[labels.for_client(d, min_size=min_size) for d in disks],
             storage_encryption=se)
 
+    def _add_structure(self, *, disk: Disk, next_offset: int, is_last: bool,
+                       structure: snapdapi.VolumeStructure):
+        ptype = structure.type.split(',', 1)[1].upper()
+        if structure.offset is not None:
+            offset = structure.offset
+        else:
+            offset = next_offset
+        if structure.role == snapdapi.Role.SYSTEM_DATA and is_last:
+            size = gaps.largest_gap(disk).size
+        else:
+            size = structure.size
+        next_offset = offset + size
```

Comment:
Please tell me this is gpt only.

Reply:
this comment prompted me to refactor how we make the assertions to need to about the system we're installing. and yes, gpt only!

Reply:
:tada: